lib/rules/no-restricted-imports.js

```diff
@@ -400,6 +467,15 @@ module.exports = {
 							restrictedImportNames.includes(importName)
 						) {
 							specifiers.forEach(specifier => {
+								// Skip if this is a type-only import specifier and type imports are allowed
+								if (
+									allowTypeImports &&
+									specifier.specifier &&
+									specifier.specifier.importKind === "type"
+								) {
+									return;
+								}
```

Comment:
This looks like a false positive:

```ts
/* eslint no-restricted-imports: ["error", {
  "paths": [{
    "name": "foo",
	"importNames": ["bar"]
	"allowTypeImports": true
  }]
}]
*/

export { type bar, baz } from "foo";

```

(the same for the other four places where only `specifier.specifier.importKind` is checked).

Reply:
Fixed in https://github.com/eslint/eslint/pull/19562/commits/59711edf410e6ebc6b4d3d729b6735b890addec1